metricbeat/module/system/process/process_test.go

```diff
@@ -37,13 +38,17 @@ func TestFetch(t *testing.T) {
 
 	f := mbtest.NewReportingMetricSetV2Error(t, getConfig())
 	events, errs := mbtest.ReportingFetchV2Error(f)
-	assert.Empty(t, errs)
+	for _, err := range errs {
+		assert.True(t, errors.Is(err, process.NonFatalErr{}))
```

Comment:
Why not use `assert.ErrorIs`?
The problem of this assert is that you won't see any context in the output, it will be just something like "expected true but false". We're trying to avoid error messages like that. The more specialized assert will do a better job.

Reply:
@rdner thanks for this!
I've updated the assertion.